### PR TITLE
Adds all comment deletion support

### LINKED ISSUE #33: Add delete all comments button
- [ ] Add a button in `comments.html` and style in `comments-style.css`
- [ ] Send post request to delete all entities

@billie-cat @f-camp

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request adds a feature allowing any user to delete all the comments in the database. Pull request adds a new file `DeleteSerlet.java` which queries the database and removes all comments.

Screenshots: https://screenshot.googleplex.com/VtvdmRuEfKc

### CODE REVIEW

**Comment by f-camp** [comment]
Please can you elaborate on the delete feature, can the user delete ALL the comments in the db or just theirs? Why is something we want?

**Comment by jaysaleh** [comment]
It deletes all comments in the database. Below is a screenshot from the walk through:

https://screenshot.googleplex.com/wynA5G6Owe5.png

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  private static final String COMMENT = "Comment";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(COMMENT);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
```
Comment (f-camp) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: This be replaced with datastore.delete(results.stream().map(entity->entity.getKey()).collect(toList())?

Please take a look at compiling the keys into an Iterable and using https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete-java.lang.Iterable-.

Ideally we should use a transaction I feel so that we can rollback if there was an issue deleting one key in the database. As this is written it is possible to continue deleting and not be sure if we deleted all or not, using a transaction will help with that. Perhaps the .delete(Iterable<>keys) would help with that I'm not sure. 
https://www.justsoftwaresolutions.co.uk/database/database_tip_use_transactions.html

Also take a look at Java lambdas and streams, this will help you in the future.
https://winterbe.com/posts/2014/07/31/java8-stream-tutorial-examples/
https://www.infoworld.com/article/3452018/get-started-with-lambda-expressions.html

**Review by f-camp — COMMENTED** [review]
Comments added.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java (hunk 1 above)
Comment (jaysaleh) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: Added as a TODO

File: portfolio/src/main/webapp/script/comment-script.js [hunk 2]
```diff
@@ -33,6 +33,16 @@ async function getData() {
   }
 }
 
+/**
+ * Forwards POST request to delete-data servlet and refreshes portfolio with
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script/comment-script.js]: "Forward POST request to delete-data servlet" explains what the code is doing and not what the method does. I would rephrase as:

Deletes comments and refreshes portfolio with updated comments.

**Review by billie-cat — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script/comment-script.js (hunk 2 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script/comment-script.js]: done

**Review by f-camp — APPROVED** [review]

**Review by billie-cat — APPROVED** [review]
